src/components/Collapse.svelte

```diff
@@ -0,0 +1,145 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
```

Comment:
How hard would it be to break up these `details` -> `Collapse` changes into their own commit/PR?

Reply:
Not entirely sure though it could take a decent chunk of work. Thought about splitting that work into a separate PR but this refactor heavily influenced the Collapse component design so kept it in for ease of iteration.

Reply:
Or are you maybe thinking that we split out the changes in other files that aren't involved in constraints/scheduling goals/scheduling conditions into a follow-up PR?